src/main/java/org/jetbrains/research/refactorinsight/folding/handlers/MoveOperationFoldingHandler.java

```diff
@@ -0,0 +1,73 @@
+package org.jetbrains.research.refactorinsight.folding.handlers;
+
+import com.intellij.psi.PsiCodeBlock;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.research.refactorinsight.adapters.RefactoringType;
+import org.jetbrains.research.refactorinsight.data.RefactoringInfo;
+import org.jetbrains.research.refactorinsight.utils.PsiUtils;
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+public class MoveOperationFoldingHandler implements FoldingHandler {
+  @NotNull
+  @Override
+  public List<Folding> getFolds(@NotNull RefactoringInfo info, @NotNull PsiFile file, boolean isBefore) {
+    PsiMethod method = PsiUtils.findMethod(file, isBefore ? info.getNameBefore() : info.getNameAfter());
+    if (method == null) {
+      return Collections.emptyList();
+    }
+    String details = isBefore ? info.getLeftPath() : info.getRightPath();
+    String hintText = specificOperation(info.getType())
+        + (isBefore ? " to " : " from ")
+        + details.substring(details.lastIndexOf(File.separatorChar) + 1)
+        + (info.isChanged() ? " with changes" : " without changes");
+    PsiCodeBlock body = method.getBody();
+    return Collections.singletonList(
+        new Folding(
+            hintText,
+            method.getTextRange().getStartOffset(),
+            body == null ? -1 : body.getTextRange().getStartOffset(),
+            body == null ? -1 : body.getTextRange().getEndOffset()
+        )
+    );
+  }
+
+  @NotNull
+  @Override
+  public Folding uniteFolds(@NotNull List<Folding> folds) {
+    String operation;
+    if (folds.stream().allMatch(folding -> folding.hintText.startsWith("Moved"))) {
+      operation = "Moved";
+    } else if (folds.stream().allMatch(folding -> folding.hintText.startsWith("Pulled up"))) {
+      operation = "Pulled up";
+    } else if (folds.stream().allMatch(folding -> folding.hintText.startsWith("Pushed down"))) {
+      operation = "Pushed down";
+    } else {
+      throw new AssertionError("Folds of different types");
+    }
+    String hintText = folds.stream().allMatch(folding -> folding.hintText.contains("without changes"))
```

Comment:
make it boolean, and then add "without changes" to hintText if true (all folds match)